autosportlabs/racecapture/views/analysis/sessionlistview.py

```
@@ -147,34 +142,62 @@ def __init__(self, **kwargs):
         sv.add_widget(accordion)
         self._accordion = accordion
         self.add_widget(sv)
+        self.sessions = []
+        self.datastore = None
+        self.settings = None
+        self._save_timeout = None
+        self._session_accordion_items = []
+
+    def init_view(self):
+        if self.settings and self.datastore:
+            selection_settings_json = self.settings.userPrefs.get_pref('analysis_preferences', 'selected_sessions_laps')
+            selection_settings = json.loads(selection_settings_json)
+            delete_sessions = []
+
+            # Load sessions first, then select the session
+            for session_id_str, session_info in selection_settings["sessions"].iteritems():
+                session = self.datastore.get_session_by_id(int(session_id_str))
+                if session:
+                    self.append_session(session)
+                else:
+                    # If the session doesn't exist anymore, remove it from settings
+                    delete_sessions.append(session_id_str)
+
+            if len(delete_sessions) > 0:
+                for session_id in delete_sessions:
+                    selection_settings["sessions"].pop(session_id)
+
+                # Resave loaded sessions and laps
+                self._save()
+
+            for session_id_str, session_info in selection_settings["sessions"].iteritems():
+                session_id = int(session_id_str)
+
+                laps = session_info["selected_laps"]
+
+                for lap in laps:
+                    self.select_lap(session_id, lap, True)
+
+        else:
+            Logger.error("SessionListView: init_view failed, missing settings or datastore object")
```

should raise an exception here so crash handler can report it?